source/CMakeLists.txt

```diff
@@ -34,6 +34,37 @@ add_library(
     input_conv.cpp
 )
 
+list(APPEND device_srcs
+    module_hamilt/src/nonlocal.cpp
+    module_hamilt/src/veff.cpp
+    module_hamilt/src/ekinetic.cpp
+    module_pw/src/pw_multi_device.cpp
+    module_hsolver/src/dngvd_op.cpp
+    module_hsolver/src/math_kernel.cpp
+    module_elecstate/src/elecstate_multi_device.cpp
+    module_psi/src/memory_psi.cpp
+    module_psi/src/device.cpp
+)
+
+if(USE_CUDA)
+  list(APPEND device_srcs
+      module_hamilt/src/cuda/nonlocal.cu
+      module_hamilt/src/cuda/veff.cu
+      module_hamilt/src/cuda/ekinetic.cu
+      module_pw/src/cuda/pw_multi_device.cu
+      module_hsolver/src/cuda/dngvd_op.cu
+      module_hsolver/src/cuda/math_kernel.cu
+      module_elecstate/src/cuda/elecstate_multi_device.cu
+      module_psi/src/cuda/memory.cu
+  )
+endif()
+
+add_library(device OBJECT ${device_srcs})
+
+if(USE_CUDA)
+  target_link_libraries(device cusolver cublas cufft)
```

Comment:
Where were those CUDA libraries found? Maybe by `enable_language(CUDA)`? If you link them here, I think it's duplicate to link them in the root CMakeLists.txt .

Reply:
Agree, I have deleted the redundant links.